### PR TITLE
Feature: adds a Ubuntu 16.04 build target

### PR DESCRIPTION
**Pull request by lsh-0**

### CODE REVIEW

**Comment by lsh-0** [comment]
another 'host name too long' error, 67 chars this time: 

> journal-formula-pr-88-fresh-critical-css--journal.elifesciences.org

so, it's going to need to be shorter, can't get around the 63 byte limit.

I don't think the hashed hostnames feature of openssh/ssh-keygen (the only thing failing right now) is the right fit for this problem.

Hashing the hostname outside of openssh might be possible and guarantee a fix for this. We could refer to the hash with an alias. Terrible readability though and I'm not sure what depends on hostnames - definitely nginx I'm thinking. 

Could we truncate it to just `journal-formula-pr-88-fresh-critical-css` ?

**Comment by lsh-0** [comment]
I'm looking at the journal formula now and it has 14.04 specific logic and I don't see any evidence of a 16.04 upgrade branch, so it's getting the 1804 treatment. I'll change the build target to `s1804`

File: salt/journal/init.sls [hunk 1]
```diff
@@ -6,6 +8,20 @@ maintenance-mode-start:
         - require:
             - nginx-server-service
 
+{% if osrelease not in ["14.04", "16.04"] %}
```
Comment (lsh-0) [inline salt/journal/init.sls]: probably not at all necessary if the journal instance switches to containers like Chris mentioned

**Comment by lsh-0** [comment]
I'm building locally on 16.04 vagrant now.

Had some problems with `npm` missing in 18.04, so I added it to the `nodejs6` builder-base formula (locally). Not sure where the best place for it is. It might not even be necessary for 16.04

**Comment by giorgiosironi** [comment]
> so, it's going to need to be shorter, can't get around the 63 byte limit.

Had this problem before, the variants get longer and longer (`elife-libraries-formula-pr-...`) until the hostname doesn't fit anymore. It does need to be shortened outside of that state, for consistency (and because the DNS entry will be created outside of Salt and must match). The `s1804` alias for `standalone1804` comes from this.

**Comment by giorgiosironi** [comment]
> `journal-formula-pr-88-fresh-critical-css--journal.elifesciences.org`

This should be `journal-pr-...`, not `journal-formula`, possibly. There is an optimization that is not firing in `elifeFormula` step.

**Comment by giorgiosironi** [comment]
By the way the best option here should be to port the formula to use the [existing container image](https://github.com/elifesciences/journal/blob/develop/Dockerfile) first, which will greatly simplify the migration. So I'd wait on this formula change, favoring the other projects that are already on the EC2+container approach or that don't have a container image at all.

**Pull request closed without merging**